eth/consensus/clique/encoding.py

```diff
@@ -0,0 +1,136 @@
+import rlp
+from typing import (
+    Tuple,
+)
+
+from eth_typing import (
+    Address,
+)
+
+from eth.rlp.sedes import (
+    uint256,
+)
+from eth.consensus.clique.datatypes import (
+    Snapshot,
+    Tally,
+    Vote,
+    VoteAction,
+)
+
+
+def encode_address_tally_pair(pair: Tuple[Address, Tally]) -> bytes:
+    return rlp.encode(
+        [pair[0], encode_tally(pair[1])],
+        sedes=rlp.sedes.List(
+            (rlp.sedes.binary, rlp.sedes.binary)
+        ),
+    )
+
+
+def decode_address_tally_pair(pair: bytes) -> Tuple[Address, Tally]:
+    address, tally_bytes, = rlp.decode(
+        pair,
+        rlp.sedes.List((rlp.sedes.binary, rlp.sedes.binary))
+    )
+
+    tally = decode_tally(tally_bytes)
+
+    return address, tally
+
+
+def encode_vote(vote: Vote) -> bytes:
+    return rlp.encode(
+        [
+            vote.signer,
+            vote.block_number,
+            vote.subject,
+            vote.action.value,
+        ],
+        sedes=rlp.sedes.List(
+            (
+                rlp.sedes.binary,
+                uint256,
+                rlp.sedes.binary,
+                rlp.sedes.binary,
+            )
+        ),
+    )
+
+
+def decode_vote(vote: bytes) -> Vote:
+    signer, block_number, subject, action = rlp.decode(
+        vote,
+        rlp.sedes.List((rlp.sedes.binary, uint256, rlp.sedes.binary, rlp.sedes.binary))
+    )
+    return Vote(
+        signer=signer,
+        block_number=block_number,
+        subject=subject,
+        action=VoteAction.NOMINATE if action == VoteAction.NOMINATE.value else VoteAction.KICK
+    )
+
+
+def encode_snapshot(snapshot: Snapshot) -> bytes:
+    return rlp.encode(
+        [
+            snapshot.block_hash,
+            [signer for signer in snapshot.signers],
+            [encode_vote(vote) for vote in snapshot.votes],
+            [
+                encode_address_tally_pair((address, tally))
+                for address, tally in snapshot.tallies.items()
+            ],
+        ],
+        sedes=rlp.sedes.List(
```

Comment:
this code should be easier to read if these `sedes` values are set as constants at the module level.